### PR TITLE
SPOI-10218 Implementation of Kinesis to S3 application template

### PR DESCRIPTION
**Pull request by chaithu14**
@yogidevendra Please review.

### CODE REVIEW

File: kinesis-to-s3/src/main/java/com/datatorrent/apps/Application.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+@ApplicationAnnotation(name="kinesis-to-S3")
+public class Application implements StreamingApplication
+{
+  @Override
+  public void populateDAG(DAG dag, Configuration conf)
+  {
+    // Add Kinesis as input and S3 as output operators respectively to dag.
+    KinesisByteArrayInputOperator inputModule = dag.addOperator("KinesisInput", new KinesisByteArrayInputOperator());
+    S3TupleOutputModule.S3BytesOutputModule outputModule = dag.addModule("S3OutputModule", new S3TupleOutputModule.S3BytesOutputModule());
```
Comment (yogidevendra) [inline kinesis-to-s3/src/main/java/com/datatorrent/apps/Application.java]: Add commented code block to apply transformations in between.

File: kinesis-to-s3/src/main/java/com/datatorrent/apps/KinesisByteArrayInputOperator.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ */
+package com.datatorrent.apps;
+
+import java.nio.ByteBuffer;
+
+import com.amazonaws.services.kinesis.model.Record;
+
+import com.datatorrent.contrib.kinesis.AbstractKinesisInputOperator;
+
+public class KinesisByteArrayInputOperator extends AbstractKinesisInputOperator<byte[]>
```
Comment (yogidevendra) [inline kinesis-to-s3/src/main/java/com/datatorrent/apps/KinesisByteArrayInputOperator.java]: Malhar should have concrete implementation for byte[]

File: kinesis-to-s3/src/main/resources/META-INF/properties.xml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+-->
+<configuration>
+    <property>
+      <name>dt.operator.KinesisInput.prop.accessKey</name>
+      <value>ACCESSKEY</value>
```
Comment (yogidevendra) [inline kinesis-to-s3/src/main/resources/META-INF/properties.xml]: Could you please make sure to have default values which are consistent with other apps.

File: kinesis-to-s3/README.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+### Description
+The Kinesis S3 Application Template continuously ingest messages from Kinesis and upload those to Amazon S3.
+- The application scales linearly with number of Kinesis shards.
+- The application is fault tolerant and can withstand node and cluster outages without data loss.
+- The application is also highly performance and can perform as fast as the network bandwidth allows.
+- The only configuration user needs to provide is source Kinesis credentials, topic name and destination S3 bucket name, path and credentials.
+- This enterprise grade application template will dramatically reduce your time to market and cost of operations.
```
Comment (yogidevendra) [inline kinesis-to-s3/README.md]: Could you please make this README consistent with other apps?

File: kinesis-to-s3/src/main/java/com/datatorrent/apps/KinesisByteArrayInputOperator.java (hunk 2 above)
Comment (chaithu14) [inline kinesis-to-s3/src/main/java/com/datatorrent/apps/KinesisByteArrayInputOperator.java]: Created JIRA for the same: APEXMALHAR-2431
I will remove it from here once the APEXMALHAR-2431 JIRA is fixed.

File: kinesis-to-s3/README.md (hunk 4 above)
Comment (chaithu14) [inline kinesis-to-s3/README.md]: I will open a separate PR for the same.
JIRA for tracking this task is below:
https://datatorrent.atlassian.net/browse/SPOI-10221

File: kinesis-to-s3/pom.xml [hunk 5]
```diff
@@ -0,0 +1,326 @@
...
+  <artifactId>kinesis-to-s3</artifactId>
+  <packaging>jar</packaging>
+
+  <!-- change these to the appropriate values -->
+  <name>Kinesis S3 App</name>
+  <description>Ingest messages from Kinesis to Amazon S3.</description>
+
+  <properties>
+    <!-- change this if you desire to use a different version of Apex Core -->
+    <apex.version>3.5.0-SNAPSHOT</apex.version>
```
Comment (chaithu14) [inline kinesis-to-s3/pom.xml]: Done.

File: kinesis-to-s3/src/main/java/com/datatorrent/apps/Application.java (hunk 1 above)
Comment (chaithu14) [inline kinesis-to-s3/src/main/java/com/datatorrent/apps/Application.java]: Done.

File: kinesis-to-s3/src/main/resources/META-INF/properties.xml (hunk 3 above)
Comment (chaithu14) [inline kinesis-to-s3/src/main/resources/META-INF/properties.xml]: Done.

**Pull request merged**